apps/condo/domains/organization/integrations/sbbol/utils/getAccessTokenForUser.js

```diff
@@ -24,66 +24,36 @@ async function initializeSbbolAuthApi (useExtendedConfig = false) {
     }
-
-    if (await sbbolSecretStorage.isAccessTokenExpired(userId)) {
-        const clientSecret = await sbbolSecretStorage.getClientSecret()
-        const currentRefreshToken = await sbbolSecretStorage.getRefreshToken(userId)
-        const oauth2 = new SbbolOauth2Api({ clientSecret, useExtendedConfig })
+    const oauth2 = await initializeSbbolAuthApi(useExtendedConfig)
+    if (await sbbolSecretStorage.isAccessTokenExpired(userId, organizationId)) {
+        const currentRefreshToken = await sbbolSecretStorage.getRefreshToken(userId, organizationId)
         const { access_token, expires_at: expiresAt, refresh_token } = await oauth2.refreshToken(currentRefreshToken)
-
-        await sbbolSecretStorage.setAccessToken(access_token, userId, { expiresAt })
-        await sbbolSecretStorage.setRefreshToken(refresh_token, userId)
+        await sbbolSecretStorage.setAccessToken(access_token, userId, organizationId, { expiresAt })
+        await sbbolSecretStorage.setRefreshToken(refresh_token, userId, organizationId)
     }
-
-    return await sbbolSecretStorage.getAccessToken(userId)
+    const { accessToken, ttl } = await sbbolSecretStorage.getAccessToken(userId, organizationId)
+    return { error: null, accessToken, ttl }
 }
 
 const getAllAccessTokensByOrganization = async (context, organizationId) => {
     const employees = await find('OrganizationEmployee', {
         organization: { id: organizationId },
         deletedAt: null,
     })
-
-    let accessTokens = []
-    const sbbolSecretStorage = getSbbolSecretStorage(true)
-    const clientSecret = await sbbolSecretStorage.getClientSecret()
-    const oauth2 = new SbbolOauth2Api({ clientSecret, useExtendedConfig: true })
-
-    for (let employee of employees) {
-        const userId = employee.user
-        let accessToken
-        try {
-            if (await sbbolSecretStorage.isRefreshTokenExpired(userId)) {
-                continue
-            }
-
-            if (await sbbolSecretStorage.isAccessTokenExpired(userId)) {
-                const currentRefreshToken = await sbbolSecretStorage.getRefreshToken(userId)
-                const { access_token, expires_at: expiresAt, refresh_token } = await oauth2.refreshToken(currentRefreshToken)
-                await sbbolSecretStorage.setAccessToken(access_token, userId, { expiresAt })
-                await sbbolSecretStorage.setRefreshToken(refresh_token, userId)
-                accessToken = access_token
-            } else {
-                accessToken = await sbbolSecretStorage.getAccessToken(userId)
-            }
-
-            accessTokens.push(accessToken)
-        } catch (e) {
-            // continue finding accessTokens
-        }
-    }
-
-    return accessTokens
+    const accessTokens = await Promise.all(employees.map(async ({ user: userId }) => {
+        const { error, ...accessToken } = await getAccessTokenForUser(userId, organizationId, true)
+        return error ? null : accessToken
```

Comment:
Who is this error for? Do we need to log it?

Reply:
this function is not used